src/components/shared/Navbar.tsx

```diff
@@ -61,10 +65,91 @@ const NavbarUserSection = styled.div`
     }
 `;
 
+const userImage = css`
+    width: 40px;
+    height: 40px;
+    border-radius: 50%;
+`;
+
+const userBox = css`
+    display: flex;
+    align-items: center;
+    cursor: pointer;
+    position: relative;
+`;
+
+const modalContent = css`
+    background: #fff;
+    padding: 10px;
+    border-radius: 5px;
+    box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+    position: absolute;
+    top: 150%;
+    right: 0;
+    min-width: 200px;
+    display: flex;
+    flex-direction: column;
+    z-index: 1000;
+`;
+
+const modalLink = css`
+    margin: 5px 0;
+    color: #1f2937;
+    text-decoration: none;
+    cursor: pointer;
+    &:hover {
+        color: #4b5563;
+    }
+`;
+
 function Navbar() {
-    // Grab user from user global state.
-    const user = useAppSelector((state) => state.setUser);
-    const userNameCapitalized = grabUserNameCapitalized(user);
+    const userId = useAppSelector((state) => state.user.userId);
+    const [user, setUser] = useState<User | null>();
+    const [modalOpen, setModalOpen] = useState(false);
+    const modalRef = useRef<HTMLDivElement | null>(null);
+    const navigate = useNavigate();
+
+    const toggleModal = () => {
+        setModalOpen(!modalOpen);
+    };
+
+    // If the modal is open and the click event's target is not within the modal, close the modal
+    const handleClickOutside = (event) => {
+        if (modalRef.current && !modalRef.current.contains(event.target)) {
+            setModalOpen(false);
+        }
+    };
+
+    const navigateTo = (path) => {
+        navigate(path);
+        setModalOpen(false);
+    };
+
+    const handleSignOut = () => {
+        signOut();
+        setModalOpen(false);
+        navigate('/');
+    }
+
+    useEffect(() => {
+        const fetchUser = async () => {
+            if(userId) {
+                const userRef = await getUser(userId);
+                if(userRef) {
+                    setUser(userRef);
+                }
+            }
+        }
+        fetchUser();
+    }, [userId]);
+
```

Comment:
Here’s how to get the userId and query Firestore from an individual component (Part 2)

- Query Firestore using the userId with useEffect
- The getUser and setUser function is stored in `/src/features/user/userAPI`